### PR TITLE
411 front export functionality

### PR DESCRIPTION
**Pull request by jmensch1**
This adds a PNG export for the map; PNG, CSV, and PDF exports for the individual charts; and a multi-page PDF export for all of the charts.

The map export is a little funky, code-wise. The leaflet-easy-print package needed a fixed height and width for the map for the export to work, hence the DOM measuring stuff. And in order to use our own button, I had to make their button invisible and then programmatically click it.

This PR also includes some other minor stuff: 
- fixed some console errors that proptypes was throwing
- added a show more/show less toggle to Criteria, since it looked bad when you clicked "all" and then all of the council names appeared in the box.
- added live data to Frequency and Total Requests charts

### CODE REVIEW

**Comment by jmensch1** [comment]
hi @sellnat77, I'm getting a Terraform error, something about security credentials -- 

Error: error using credentials to get account ID: error calling sts:GetCallerIdentity: InvalidClientTokenId: The security token included in the request is invalid.

It's weird cuz I had a successful PR a few days ago...any idea what might be happening? Thanks much

**Comment by sellnat77** [comment]
Long story short, creds were leaked on a diff project so Joel is revoking all access
We are going to shoot for the heroku route now

File: src/components/PinMap/PinMap.jsx [hunk 1]
```diff
@@ -215,14 +242,34 @@ class PinMap extends Component {
...
             </Overlay>
           </LayersControl>
+          <PrintControl
+            sizeModes={['Current']}
+            hideControlContainer={false}
+            exportOnly
+          />
         </Map>
-      </div>
+        <button
```
Comment (brodly) [inline src/components/PinMap/PinMap.jsx]: can we use our fancy Button component?

File: src/components/Visualizations/Criteria.jsx [hunk 2]
```diff
@@ -1,18 +1,25 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'proptypes';
 import { connect } from 'react-redux';
 
+// if more than this, use the more/less toggle
+const MAX_COUNCILS = 10;
```
Comment (brodly) [inline src/components/Visualizations/Criteria.jsx]: Thanks, I was going to suggest something like this

File: src/components/Visualizations/Criteria.jsx [hunk 3]
```diff
@@ -30,6 +37,18 @@ const Criteria = ({
             Neighborhood Council District
           </span>
           <span>{ councilsText }</span>
+          { councils.length > MAX_COUNCILS && (
+            <span>
+              <span>{ showAll ? '' : '...' }</span>
+              <button
```
Comment (brodly) [inline src/components/Visualizations/Criteria.jsx]: Button component

**Comment by jmensch1** [comment]
Thanks Josh, I made those changes

**Pull request merged**